### PR TITLE
Allows pAI to be emagged

### PR DESCRIPTION
**Pull request by Randomguy523**
## About The Pull Request

Allows pAIs that have a master already set to be emagged to have their master string removed, freeing the pAI to be bound to someone else, likely whoever emagged the pAI.

## Why It's Good For The Game

This is something some players thought was possible for a long while, but it wasn't, now pAIs are prone to emags like the rest of the silicons.

Also addresses #56849
## Changelog
:cl:
add: pAIs can now be emagged
/:cl:

### CODE REVIEW

**Comment by TheBonded** [comment]
now we just need a cool emag program like a nuke disk tracker (from emagging tablets) or a chameleon PDA messanger

**Review by Timberpoes — CHANGES_REQUESTED** [review]
The pAI would also benefit from having its supplemental laws wiped when you do this.

You can do this very simply by calling `pai.clear_supplied_laws()`

**Comment by Timberpoes** [comment]
When that's done, this is approved for merging. I also thought this was possible. I bet some old wiki entry that was out of date.

**Comment by Randomguy523** [comment]
Turns out `pai.clear_supplied_laws()` breaks the supplemental entirely, unless that would be a desired effect.

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 1]
```diff
@@ -327,3 +327,13 @@
...
+
+/obj/item/paicard/emag_act(mob/user) // Emag to wipe the master DNA and supplemental directive
+	if(pai)
+		to_chat(user, "<span class='notice'>You override [pai]'s directive system, clearing its master string and supplied directive.</span>")
+		to_chat(pai, "<span class='danger'>Warning: System override detected, check directive sub-system for any changes.'</span>")
+		log_game("[key_name(user)] emagged [key_name(pai)], wiping their master DNA and supplemental directive.")
+		pai.master = null
+		pai.master_dna = null
+		pai.laws.supplied[1] = "None." // Set supplemental directive to this
+	return
```
Comment (Mothblocks) [inline code/modules/mob/living/silicon/pai/pai.dm:339]:
```suggestion
```

Trailing returns are not necessary.

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 2]
```diff
@@ -327,3 +327,13 @@
 		return
 
 	return ..()
+
+/obj/item/paicard/emag_act(mob/user) // Emag to wipe the master DNA and supplemental directive
+	if(pai)
```
Comment (Mothblocks) [inline code/modules/mob/living/silicon/pai/pai.dm:332]: Make this an early return, as in:

```dm
if (!pai)
    return
```

**Comment by Mothblocks** [comment]
Undraft when this is ready.

**Pull request merged**